File: ios/MedPharm/MedPharm/MedPharmApp.swift
/*
 * MedPharm ERP - iOS Application
 * Copyright (C) 2026 Enlightec Ltd. (www.enlightec.com)
 * Author: Robert Andrew Stillwell
 * Email: Andrew.Stillwell@enlightec.com
 * License: GNU General Public License v3.0
 *
 * SwiftUI iOS client for MedPharm Cloud API.
 * Requires: iOS 17+, Xcode 15+
 */

import SwiftUI

@main
struct MedPharmApp: App {
    @StateObject private var authManager = AuthManager()

    var body: some Scene {
        WindowGroup {
            if authManager.isAuthenticated {
                MainTabView()
                    .environmentObject(authManager)
            } else {
                LoginView()
                    .environmentObject(authManager)
            }
        }
    }
}

File: android/app/src/main/java/com/enlightec/medpharm/MedPharmApp.kt
/*
 * MedPharm ERP - Android Application
 * Copyright (C) 2026 Enlightec Ltd. (www.enlightec.com)
 * Author: Robert Andrew Stillwell
 * Email: Andrew.Stillwell@enlightec.com
 * License: GNU General Public License v3.0
 */

package com.enlightec.medpharm

import android.app.Application
import com.enlightec.medpharm.data.api.ApiClient

class MedPharmApp : Application() {

    override fun onCreate() {
        super.onCreate()
        ApiClient.init(this)
    }
}
